### ISSUE TITLE
0.6.0: pep517 based build fails with `Invalid version: 'UNKNOWN'` error message

### ISSUE THREAD

**Issue by kloczek** [RKrahl/pytest-dependency#76]
```console
+ /usr/bin/python3 -sBm build -w --no-isolation
* Getting build dependencies for wheel...
warning: cannot determine version number
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
    main()
  File "/usr/lib/python3.8/site-packages/pyproject_hooks/_in_process/_in_process.py", line 335, in main
    json_out['return_val'] = hook(**hook_input['kwargs'])
  File "/usr/lib/python3.8/site-packages/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
    return hook(config_settings)
  File "/usr/lib/python3.8/site-packages/setuptools/build_meta.py", line 325, in get_requires_for_build_wheel
    return self._get_build_requires(config_settings, requirements=['wheel'])
  File "/usr/lib/python3.8/site-packages/setuptools/build_meta.py", line 295, in _get_build_requires
    self.run_setup()
  File "/usr/lib/python3.8/site-packages/setuptools/build_meta.py", line 480, in run_setup
    super(_BuildMetaLegacyBackend, self).run_setup(setup_script=setup_script)
  File "/usr/lib/python3.8/site-packages/setuptools/build_meta.py", line 311, in run_setup
    exec(code, locals())
  File "<string>", line 124, in <module>
  File "/usr/lib/python3.8/site-packages/setuptools/__init__.py", line 103, in setup
    return distutils.core.setup(**attrs)
  File "/usr/lib/python3.8/site-packages/setuptools/_distutils/core.py", line 147, in setup
    _setup_distribution = dist = klass(attrs)
  File "/usr/lib/python3.8/site-packages/setuptools/dist.py", line 314, in __init__
    self.metadata.version = self._normalize_version(self.metadata.version)
  File "/usr/lib/python3.8/site-packages/setuptools/dist.py", line 350, in _normalize_version
    normalized = str(Version(version))
  File "/usr/lib/python3.8/site-packages/setuptools/_vendor/packaging/version.py", line 198, in __init__
    raise InvalidVersion(f"Invalid version: '{version}'")
setuptools.extern.packaging.version.InvalidVersion: Invalid version: 'UNKNOWN'
```

**Comment by RKrahl** [comment]
I have no idea what you where trying to do. I'd need more information in order to give a somewhat useful answer.

```
$ tar xfz pytest-dependency-0.6.0.tar.gz
$ cd pytest-dependency-0.6.0
$ python3.8 -sBm build -w --no-isolation
* Getting build dependencies for wheel...
running egg_info
creating src/pytest_dependency.egg-info
writing src/pytest_dependency.egg-info/PKG-INFO
writing dependency_links to src/pytest_dependency.egg-info/dependency_links.txt
writing entry points to src/pytest_dependency.egg-info/entry_points.txt
writing requirements to src/pytest_dependency.egg-info/requires.txt
writing top-level names to src/pytest_dependency.egg-info/top_level.txt
writing manifest file 'src/pytest_dependency.egg-info/SOURCES.txt'
adding license file 'LICENSE.txt' (matched pattern 'LICEN[CS]E*')
reading manifest file 'src/pytest_dependency.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'src/pytest_dependency.egg-info/SOURCES.txt'
* Building wheel...
running bdist_wheel
running build
running build_py
running meta
version: 0.6.0
creating build
creating build/lib
copying (with substitutions) src/pytest_dependency.py -> build/lib
installing to build/bdist.linux-x86_64/wheel
running install
running install_lib
creating build/bdist.linux-x86_64
creating build/bdist.linux-x86_64/wheel
copying build/lib/pytest_dependency.py -> build/bdist.linux-x86_64/wheel
running install_egg_info
running egg_info
writing src/pytest_dependency.egg-info/PKG-INFO
writing dependency_links to src/pytest_dependency.egg-info/dependency_links.txt
writing entry points to src/pytest_dependency.egg-info/entry_points.txt
writing requirements to src/pytest_dependency.egg-info/requires.txt
writing top-level names to src/pytest_dependency.egg-info/top_level.txt
adding license file 'LICENSE.txt' (matched pattern 'LICEN[CS]E*')
reading manifest file 'src/pytest_dependency.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'src/pytest_dependency.egg-info/SOURCES.txt'
Copying src/pytest_dependency.egg-info to build/bdist.linux-x86_64/wheel/pytest_dependency-0.6.0-py3.8.egg-info
running install_scripts
adding license file "LICENSE.txt" (matched pattern "LICEN[CS]E*")
creating build/bdist.linux-x86_64/wheel/pytest_dependency-0.6.0.dist-info/WHEEL
creating '/home/abuild/test/pytest-dependency-0.6.0/dist/tmp8lwczho7/pytest_dependency-0.6.0-py3-none-any.whl' and adding 'build/bdist.linux-x86_64/wheel' to it
adding 'pytest_dependency.py'
adding 'pytest_dependency-0.6.0.dist-info/LICENSE.txt'
adding 'pytest_dependency-0.6.0.dist-info/METADATA'
adding 'pytest_dependency-0.6.0.dist-info/WHEEL'
adding 'pytest_dependency-0.6.0.dist-info/entry_points.txt'
adding 'pytest_dependency-0.6.0.dist-info/top_level.txt'
adding 'pytest_dependency-0.6.0.dist-info/RECORD'
removing build/bdist.linux-x86_64/wheel
Successfully built pytest_dependency-0.6.0-py3-none-any.whl
```

Works for me.

In any case:

1. Please check out the [Installation instructions](https://pytest-dependency.readthedocs.io/en/stable/install.html) for pytest-dependency.
2. whatever you want to do to install pytest-dependency, always start from the [released source distribution](https://github.com/RKrahl/pytest-dependency/releases/download/0.6.0/pytest-dependency-0.6.0.tar.gz).

**Comment by kloczek** [comment]
Did you try to use autogenerated from git tag tar ball? 🤔

**Comment by RKrahl** [comment]
I did exactly what I wrote above: I unpacked the source distribution changed into that directory and ran `python3.8 -sBm build -w --no-isolation`.

You still didn't tell me what you were trying to do. I suspect that you didn't start from the source distribution, but from the cloned git repository. This will not work.

**Comment by kloczek** [comment]
> I did exactly what I wrote above: I unpacked the source distribution changed into that directory and ran python3.8 -sBm build -w --no-isolation. (You need to have build and wheel installed for this command to work.)

Do you have git matadata in that tar ball?
Autogenerated from git tag tar ball has no .git/directory.

BTW: why not use for example `setuptool-scm` instead `setuptools` + set of custom reads from git metadata? 🤔 
It would allow as well significantly shorten setup.py.

**Comment by kloczek** [comment]
BTW:
https://github.com/RKrahl/pytest-dependency/blob/9d5e0d3b49975b358fb38b0546b04e03d5543f93/setup.py#L10-L13
`distutil`s is already deprecated.

**Comment by kloczek** [comment]
Next thing
https://github.com/RKrahl/pytest-dependency/blob/9d5e0d3b49975b358fb38b0546b04e03d5543f93/setup.py#L162
Looks like `setuptools` can be removed from that list.

**Comment by RKrahl** [comment]
In my very first comment above, I pointed you to the [Installation instructions](https://pytest-dependency.readthedocs.io/en/stable/install.html) for pytest-dependency. May I suggest you to actually read that piece of documentation?

> Do you have git matadata in that tar ball? Autogenerated from git tag tar ball has no .git/directory.

That tar ball is the source distribution that is supposed to be the starting point for any attempt to install pytest-dependency. And yes, it contains the relevant metadata.

> BTW: why not use for example `setuptool-scm` instead `setuptools` + set of custom reads from git metadata? 🤔

I use [git-props](https://github.com/RKrahl/git-props) instead of `setuptool-scm`. It is similar, but has some advantages over the latter.

> It would allow as well significantly shorten setup.py.

Bullshit!

> `distutil`s is already deprecated.

It was deprecated _in the standard library_ and has been removed from there in Python 3.12. But it is still included in `setuptools`, actually, it is the basis that `setuptools` builds upon.

> Looks like `setuptools` can be removed from that list.

It is required for the manual installation from the source distribution.

I repeat myself: you still didn't tell me what you were trying to do. You didn't provided any reproducible error description. I thus close as `invalid`.
